core/src/main/java/dev/morphia/aggregation/AggregationImpl.java

```diff
@@ -113,44 +82,37 @@ public AggregationImpl(Datastore datastore, Class<T> source, MongoCollection<T>
 
     @Override
     public Aggregation<T> autoBucket(AutoBucket bucket) {
-        addStage(bucket);
-        return this;
+        return addStage(bucket);
```

Comment:
These changes are a little noisy in this PR.  But i get it.  I'd been meaning to do this but it just never bothered me enough to clean it up.